### DISCUSSION TITLE
How to enable/disable opentelemetry tracing dynamically

### DISCUSSION THREAD

**Discussion by pankaj72981** [open-telemetry/community#1048] [Community Discussions]
Dear folks, 

How to enable/disable tracing dynamically without restarting the application service?

I tried to disable it dynamically using below APIs, it didnt work
  GlobalOpenTelemetry.set(OpenTelemetry.noop());
    GlobalOpenTelemetry.set(NoopOpenTelemetry.getInstance());

Any suggestion?

**Comment by dmathieu** [comment]
I am doing this with a custom sampler. It has an `enabled` boolean that I can switch.
Whenever that bool is false, no span gets sampled.

**Comment by pankaj72981** [comment]
Thanks @dmathieu for the response. I'm able to achieve dynamic tracing with one extra (apart from GlobalOpenTelemetry singleton instance) No-op OpenTelemetry instance and based on boolean flag corressponding opentelemetry instance is used.

**Reply by gmann42** [reply]
@pankaj72981 I am trying to do something similar
. Can you expand more on No-op instance ? Which part of the otel sdk are you using No-op ?
TracerProvider or SpanProcessor

**Reply by dmathieu** [reply]
You should be able to do it with [NewNoopTracerProvider](https://pkg.go.dev/go.opentelemetry.io/otel/trace#NewNoopTracerProvider).

Note that this tracer provider is also the default one when no SDK is configured. So doing nothing achieves the same thing.

**Comment by pankaj72981** [comment]
But it would be better if it can be implemented with single instance. Anyway I have started another discussion in below thread

https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/5972
